### DISCUSSION TITLE
Wrong language detection for VB6 repository

### DISCUSSION THREAD

**Discussion by Gagniuc** [github-linguist/linguist#5829] [Classification]
Wrong language detection in VB6 repository (Visual Basic 6.0). For instance:

https://github.com/Gagniuc/Markov-Chains-Simulation-framework
https://github.com/Gagniuc/Markov-Chains-Prediction-framework
https://github.com/Gagniuc/Discrete-Probability-Detector-in-VB6
https://github.com/Gagniuc/Micro-chart-in-VB
https://github.com/Gagniuc/Visual-Sequence-Alignment-in-VB6
https://github.com/Gagniuc/Simple-sequence-alignment-in-VB6
https://github.com/Gagniuc/Markov-Chains-The-weather
https://github.com/Gagniuc/Markov-Chains-detector-in-VB6
https://github.com/Gagniuc/Genomin
https://github.com/Gagniuc/Mapping-pixels-to-LEDs-for-the-Photon-pixel-coupling-method
https://github.com/Gagniuc/WebCam-software-sampling
https://github.com/Gagniuc/VB6-add-GUI-objects-at-runtime
https://github.com/Gagniuc/Motif-detection-VB6
https://github.com/Gagniuc/PI-laboratory-in-VB6
https://github.com/Gagniuc/PromKappa-3.0-Objective-Digital-Stains-in-VB6

A new request was made here:
https://github.com/github/linguist/issues/5824

The question is how long it will take to solve this issue? How long it usually takes?

**Comment by lildude** [comment]
> The question is how long it will take to solve this issue? How long it usually takes?

It takes as long as it takes for someone to submit a PR to add support/modify the current classification 😁. 

We rely almost exclusively on community contributions as there are so many languages out there.

Once the PR has been submitted, it'll need to be discussed, reviewed and then merged. The change will then go into GitHub the next time a release is made, which depends on my work load, which puts it at about once every 2-3 months ATM.

There may be added delays/complications with this as noted in my comment at https://github.com/github/linguist/issues/5824#issuecomment-1067754395

**Comment by Gagniuc** [comment]
Thank you.

Please remember that `.vbp` and `.vbw` extensions are unique to the VB6 language. Thus, if any VBA files are accompanied by `.vbp` and `.vbw` extensions, than all repository is Visual Basic 6.0. Just look on any of [my repositories on VB6](https://github.com/Gagniuc?tab=repositories&q=VB6&type=&language=&sort=).

In short:

VB6 - if (`.bas` and/or `.cls` and/or `.frm` and/or `.frx`) and `.vbp` and `.vbw` exist.
VBA - if (`.bas` and/or `.cls` and/or `.frm` and/or `.frx`) but **no** `.vbp` and `.vbw` exist.

VBScript - if `.vbs` exists.
VB.NET - if `.vb` exists.

Note also that the presence of any **.xlsm** file implies the presence of VBA. Also note that VB6 is the only programming language among these (it is fully compiled).

**Reply by lildude** [reply]
That's not how Linguist works. See my comment at https://github.com/github/linguist/issues/5824#issuecomment-1068201650
